core/store/src/trie/insert_delete.rs

```diff
@@ -432,86 +435,84 @@ impl Trie {
                 }
             }
         }
-        self.fix_nodes(memory, path, key_deleted)?;
+
+        let mut child_memory_usage = 0;
+        for handle in path.into_iter().rev() {
+            // First, recompute memory usage, emulating the recursive descent.
+            let TrieNodeWithSize { node, mut memory_usage } = memory.destroy(handle);
+            memory_usage += child_memory_usage;
+            memory.store_at(handle, TrieNodeWithSize { node, memory_usage });
```

Comment:
Doing `store_at()` here and the possibly again in `TrieNode::Branch` branch should cause `debug_assert` inside `store_at()` to fail?

Reply:
Yeah, as said above, `squash_node` starts from `destroy` which is the opposite of `store_at`.
It can probably be simplified, but the current convention for existing methods is that they accept `handle: StorageHandle` in `memory`, and I don't want to touch it yet.